Classes/Domain/Search/Uri/SearchUriBuilder.php

```diff
@@ -68,6 +87,22 @@ public function injectUriBuilder(UriBuilder $uriBuilder)
         $this->uriBuilder = $uriBuilder;
     }
 
+    /**
+     * @param RoutingService $routingService
+     */
+    public function injectRoutingService(RoutingService $routingService)
+    {
+        $this->routingService = $routingService;
+    }
+
+    /**
+     * @param EventDispatcher $eventDispatcher
+     */
+    public function injectEventDispatcher(EventDispatcher $eventDispatcher)
```

Comment:
You should inject the Psr\EventDispatcherInterface, as the DI takes care of choosing the right implementation for you.

Reply:
Hi @bmack ,

Die you mean `\Psr\EventDispatcher\EventDispatcherInterface` ?

Reply:
yes

Reply:
Okay, I'll change it.

Reply:
Now the interface is sued for injection.